sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/TaskDefinitionsCheck.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.gradle.checks
+
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtLambdaArgument
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtVisitorVoid
+import org.jetbrains.kotlin.resolve.calls.util.getCalleeExpressionIfAny
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.checks.AbstractCheck
+import org.sonarsource.kotlin.api.frontend.KotlinFileContext
+
+private const val REF_NAME_GROUP = "group"
+private const val REF_NAME_DESCRIPTION = "description"
+
+@Rule(key = "S6626")
+class TaskDefinitionsCheck : AbstractCheck() {
+
+    override fun visitCallExpression(expression: KtCallExpression, kotlinFileContext: KotlinFileContext) {
+        if (!isTasksRegisterCall(expression)) return
+        val lambdaArg = getLastArgumentLambdaOrNull(expression) ?: return
+        val block = lambdaArg.getLambdaExpression()?.bodyExpression ?: return
+
+        val assignmentChecker = AssignmentChecker()
+        block.acceptChildren(assignmentChecker)
+        if (assignmentChecker.hasGroup && assignmentChecker.hasDescription) return
+
+        val missing = listOf(
+            if (assignmentChecker.hasGroup) null else "\"$REF_NAME_GROUP\"",
+            if (assignmentChecker.hasDescription) null else "\"$REF_NAME_DESCRIPTION\"",
+        ).filterNotNull().joinToString(" and ")
+
+        kotlinFileContext.reportIssue(
+            expression.parent,
```

Comment:
For UX reasons, we try to avoid highlighting large pieces of code. In this case, the task implementation could actually be arbitrarily large. The trade-off we usually do is we highlight the callee name only, e.g. in this case `register`. I know it's not perfect, if you have a better idea, I'd be curious to hear it too!